internal/discovery/foreign-cluster-operator/auth.go

```diff
@@ -107,6 +107,28 @@ func (r *ForeignClusterReconciler) getRemoteClient(
 	return nil, nil
 }
 
+func (r *ForeignClusterReconciler) getRemoteTenantNamespace(fc *discoveryv1alpha1.ForeignCluster) (bool, error) {
+	if !r.useNewAuth {
+		return false, nil
+	}
+
+	remoteNamespace, err := r.identityManager.GetRemoteTenantNamespace(
+		fc.Spec.ClusterIdentity.ClusterID, fc.Status.TenantControlNamespace.Local)
+	if err != nil {
+		klog.Error(err)
+		return false, err
+	}
+
+	if fc.Status.TenantControlNamespace == nil {
```

Comment:
Is this condition plausible? Should it not be defaulted at the API level?

Reply:
Honestly I am not sure whether it is possible to avoid this as the field is in the status. Should be tested

Reply:
ok, thanks @giorio94, I will test it